### PR TITLE
Adding parallel, filtering by county, fixing CI, and R CMD check fixes 

### PR DESCRIPTION
**Pull request by 1beb**
1. Added parallel processing census_geo_api to expedite the process of connecting with the API via the get_census_api function. The implementation is setup such that there will be a county-per-thread model. Ostensibly, a user would specify their own plan, for example: library(future); plan(multisession). This has significantly decreased the total time to run the process. Currently uses furrr::future_map_dfr with a progress indicator. Reduces the processing time for a state from more than a day to less than an hour.
2. Converted print to message so that it follows stdin/out convention.
3. Setting default retries to 3. This avoids issues where minor internet instability requires a process restart.
4. Additional log messaging to identify the start of a long process.
5. Updated description to alphabetize imports, add future, furr and purr.
6. Updated roxygen docs.
7. Adding wru.png to .Rbuildignore to relagate an R CMD check note.
8. Integrated changes with hwru branch
9. Implemented piggyback for last_c, mid_c, and first_c files, temporarily using private repo "solivella/wruData"
10. Adding "use_counties = TRUE" flag such that census data is captured only for those counties that appear in `voter.file` data. 
11. Added functionality for reading CENSUS_API_KEY from env var, to align with tidycensus. Updated readme with instructions.
12. Fixes some issues with undocumented or missing arguments for a number of functions
13. Fixes tests
14. Fixes CI

### CODE REVIEW

**Comment by 1beb** [comment]
Notes: 

- In sample data (data(voters)) when running by block with surname = F
- In new branch, `data(voters)` includes a duplicate column named "last" where "surname" already exists. "first" was also added. 
- When using the new `use_counties` flag, there is an unexpected variation in results which I noted in slack. 
- 12x speed up using `use_counties` flag.

**Comment by 1beb** [comment]
@solivella ready for review.

**Review by solivella — APPROVED** [review]
Awesome. Thank you, @1beb!

File: R/race_prediction_funs.R [hunk 1]
```diff
@@ -380,186 +413,215 @@ NULL
...
+    seed = sample(1:1000, 1) 
+  )
+  ctrl$burnin <- floor(ctrl$iter / 2)
   ctrl[names(control)] <- control
-  
-  ##Preliminary Data quality checks
-  .hasData()
+
+  ## Preliminary Data quality checks
+  wru_data_preflight()
```
Comment (1beb) [inline R/race_prediction_funs.R:430]: @solivella here.

**Pull request merged**